bftengine/src/bftengine/ClientsManager.cpp

```diff
@@ -124,13 +128,16 @@ bool ClientsManager::hasReply(NodeIdType clientId, ReqId reqSeqNum) {
     throw;
   }
 
-  const auto& repliesInfo = indexToClientInfo_.at(idx).repliesInfo;
+  auto& clientInfo = indexToClientInfo_.at(idx);
+  std::unique_lock<std::mutex> lock(clientInfo.repliesInfoGuard);
```

Comment:
Do you mind wrapping this block in `{ ... }` as well? This will make it visually clear which lines need to be guarded and which don't. Thereby, it will be easier to change/add code later. Same for other places where the lock is used at the end of the function.